### PR TITLE
[CHORE ]added a reusable result extension

### PR DESCRIPTION
**Pull request by mahmoudashraf93**
reusable Result extension to map AFError to Swift.Error

### CODE REVIEW

File: SayTheirNames/Source/Dependency/Networking/NetworkRequestor.swift [hunk 1]
```diff
@@ -40,24 +40,31 @@ final class NetworkRequestor: Dependency {
...
     public func fetchJSON(_ url: String, completion: @escaping (Result<Any, Swift.Error>) -> Swift.Void) {
         let request = self.session.request(url, headers: self.headers)
         request.responseJSON(queue: self.concurrentQueue) { (response) in
-            completion(response.result.mapError({$0.swiftError}))
+            completion(response.result.swiftError)
         }
     }
 }
+// MARK: - Result + swiftError
+extension Result where Failure == AFError {
```
Comment (evilpenguin) [inline SayTheirNames/Source/Dependency/Networking/NetworkRequestor.swift:63]: Can we put this in the Units > Extension folder as `Result+STN.swift`?

File: SayTheirNames/Source/Dependency/Networking/NetworkRequestor.swift [hunk 2]
```diff
@@ -40,24 +40,31 @@ final class NetworkRequestor: Dependency {
...
+    var swiftError: Result<Success, Error> {
+        return self.mapError{$0.swiftError}
+    }
+}
 // MARK: - AFError + swiftError
 extension AFError {
```
Comment (evilpenguin) [inline SayTheirNames/Source/Dependency/Networking/NetworkRequestor.swift:69]: Can we put this in the Units > Extension folder as `Result+STN.swift`?

File: SayTheirNames/Source/Dependency/Networking/NetworkRequestor.swift [hunk 3]
```diff
@@ -40,24 +40,31 @@ final class NetworkRequestor: Dependency {
     public func fetchDecodable<T: Decodable>(_ url: String, completion: @escaping (Result<T, Swift.Error>) -> Swift.Void) {
         let request = self.session.request(url, headers: self.headers)
         request.responseDecodable(of: T.self, queue: self.concurrentQueue) { (response) in
-            DispatchQueue.mainAsync { completion(response.result.mapError({$0.swiftError})) }
+            DispatchQueue.mainAsync {             completion(response.result.swiftError)
```
Comment (evilpenguin) [inline SayTheirNames/Source/Dependency/Networking/NetworkRequestor.swift:43]: Can we fix the white space?

File: SayTheirNames/Source/Dependency/Networking/NetworkRequestor.swift (hunk 1 above)
Comment (mahmoudashraf93) [reply SayTheirNames/Source/Dependency/Networking/NetworkRequestor.swift:63]: sure

File: SayTheirNames/Source/Dependency/Networking/NetworkRequestor.swift (hunk 3 above)
Comment (mahmoudashraf93) [reply SayTheirNames/Source/Dependency/Networking/NetworkRequestor.swift:43]: sure

File: SayTheirNames/Source/Dependency/Networking/NetworkRequestor.swift (hunk 2 above)
Comment (evilpenguin) [reply SayTheirNames/Source/Dependency/Networking/NetworkRequestor.swift:65]: Whats going on with this?

Comment (mahmoudashraf93) [reply SayTheirNames/Source/Dependency/Networking/NetworkRequestor.swift:65]: > Whats going on with this?

Do you mean the AFError extension should also go to the Units > Extension folder?

Comment (evilpenguin) [reply SayTheirNames/Source/Dependency/Networking/NetworkRequestor.swift:65]: Yeah lets throw that in there as well.

Comment (evilpenguin) [reply SayTheirNames/Source/Dependency/Networking/NetworkRequestor.swift:65]: > > Whats going on with this?
> 
> Do you mean the AFError extension should also go to the Units > Extension folder?

Indeed!

Comment (mahmoudashraf93) [reply SayTheirNames/Source/Dependency/Networking/NetworkRequestor.swift:65]: Haha cool will throw it in another file now

**Pull request merged**